src/skydb.test.js

```diff
@@ -0,0 +1,154 @@
+  it("should get EntryData", async () => {
+    const receivedData = await client.db.getEntryData(publicKey, dataKey);
+    await expect(receivedData["data"]).toEqual(rawEntryData);
+  });
+});
+
+describe("db.getRawBytes", () => {
+  it("should get RawBytes", async () => {
+    const receivedData = await client.db.getRawBytes(publicKey, dataKey);
+    await expect(receivedData["dataLink"]).toEqual(dataLink);
+    await expect("[" + receivedData["data"] + "]").toEqual(RawBytesData);
+  });
+});
+
+describe("db.setJSON", () => {
+  it("should set jsonData to skydb", async () => {
+    const receivedData = await client.db.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+describe("db.getJSON", () => {
+  it("should get jsonData from skydb", async () => {
+    const receivedData = await client.db.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("db.deleteJSON", () => {
+  it("should delete jsonData on skydb", async () => {
+    await client.db.deleteJSON(privateKey, dataKey);
+  });
+});
+
+describe("db.setJSON", () => {
+  it("This is important for the next tests with dbV2", async () => {
+    const receivedData = await client.db.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+//########################
+
+describe("dbV2.getJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("dbV2.setJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.setJSON(privateKey, dataKey, data);
+
+    await expect(receivedData.dataLink).toEqual(`sia://${skylink}`);
+  });
+});
+
+describe("dbV2.getJSON", () => {
+  it("should send post request to default portal", async () => {
+    const receivedData = await client.dbV2.getJSON(publicKey, dataKey);
+
+    await expect(receivedData.data).toEqual(data);
+  });
+});
+
+describe("dbV2.setDataLink", () => {
+  it("should be a DataLink set", async () => {
+    await client.dbV2.setDataLink(privateKey, dataKey, dataLink);
+  });
+});
+
+describe("dbV2.setEntryData", () => {
+  it("should set EntryData", async () => {
+    await client.dbV2.setEntryData(privateKey, dataKey, rawEntryData);
+  });
+});
+
+describe("dbV2.getEntryData", () => {
+  it("should get EntryData", async () => {
+    const receivedData = await client.dbV2.getEntryData(publicKey, dataKey);
+    await expect(receivedData["data"]).toEqual(rawEntryData);
+  });
+});
+
+describe("dbV2.getRawBytes", () => {
+  it("should get RawBytes", async () => {
+    const receivedData = await client.dbV2.getRawBytes(publicKey, dataKey);
+    await expect(receivedData["dataLink"]).toEqual(dataLink || dataLink_dbV2);
```

Comment:
I don't think this does what you intend it to do. The `||` will make this evaluate to `dataLink`. Try opening `node` in your terminal and doing this:

```
> 3 || "asdf"
3
```

Is there a way to change this test to only expect one value?

Reply:
I'll have to look at what I've done there and improve it.
Since "yarn test" and "yarn lint" ran clean, I thought everything was OK and would work.
And I didn't think about it because of it.

Reply:
No worries, it's common for small things like this to escape notice. That's why we do code reviews. 🙂